dotenv/parser.go

```diff
@@ -123,8 +123,8 @@ loop:
 			}
 
 			return "", "", inherited, fmt.Errorf(
-				`line %d: unexpected character %q in variable name`,
-				p.line, string(rune))
+				`line %d: unexpected character %q in variable name %q`,
+				p.line, string(rune), strings.Split(string(src), "\n")[0])
```

Comment:
`src` is a string, doesn't seem a cast is necessary here

Reply:
good point!